src/Text.Analyzers/Core/CodeAnalysisDictionary.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Text.Analyzers
+{
+    /// <summary>
+    /// Source for "recognized" misspellings and "unrecognized" spellings obtained by parsing either
+    /// XML or DIC code analysis dictionaries.
+    /// </summary>
+    /// <Remarks>
+    /// <seealso href="https://docs.microsoft.com/en-us/visualstudio/code-quality/how-to-customize-the-code-analysis-dictionary?view=vs-2019"/>
+    /// </Remarks>
+    internal sealed class CodeAnalysisDictionary
+    {
+        /// <summary>
+        /// Initialize a new instance of <see cref="CodeAnalysisDictionary"/>.
+        /// </summary>
+        /// <param name="recognizedWords">Misspelled words that the spell checker will now ignore.</param>
+        /// <param name="unrecognizedWords">Correctly spelled words that the spell checker will now report.</param>
+        private CodeAnalysisDictionary(IEnumerable<string> recognizedWords, IEnumerable<string> unrecognizedWords)
+        {
+            RecognizedWords = new HashSet<string>(recognizedWords, StringComparer.OrdinalIgnoreCase);
```

Comment:
I think you will have a warning suggesting to use the `new(...)` syntax.

Reply:
With ImmutableHashSet?

Reply:
No warnings with the current code - closing